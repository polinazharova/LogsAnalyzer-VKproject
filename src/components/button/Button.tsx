import "./Button.styles.scss";

interface ButtonProps {
  children: React.ReactNode;
  type: "submit" | "reset" | "button";
  className?: string;
  disabled?: boolean;
  handleClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;
}

const Button: React.FC<ButtonProps> = ({
  children,
  type,
  className = "",
  disabled = false,
  handleClick = (): void => {},
}) => {
  return (
    <button
      className={`button ${className}`}
      type={type}
      disabled={disabled}
      onClick={handleClick}
    >
      {children}
    </button>
  );
};

export default Button;
