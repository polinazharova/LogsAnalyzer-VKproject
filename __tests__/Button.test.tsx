import React from "react";
import Button from "../src/components/button/Button";
import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";

describe("Button component", () => {
  test("renders children correctly", () => {
    render(<Button type="submit">Upload</Button>);
    expect(screen.getByText("Upload")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { container } = render(
      <Button type="submit" className="test" disabled={true}>
        Upload
      </Button>,
    );
    expect(container).not.toBeEmptyDOMElement();
    expect(container.firstChild).toMatchSnapshot();
  });

  test("applies correct className", () => {
    render(
      <Button type="submit" className="btn">
        Class
      </Button>,
    );
    const button = screen.getByRole("button");
    expect(button).toHaveClass("btn");
  });

  test("calls onClick handler when clicked", () => {
    const handleClick = jest.fn();
    render(
      <Button type="submit" handleClick={handleClick}>
        Click
      </Button>,
    );
    fireEvent.click(screen.getByText("Click"));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
