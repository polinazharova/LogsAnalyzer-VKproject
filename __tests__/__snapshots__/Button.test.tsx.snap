// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component matches snapshot 1`] = `
<button
  class="button test"
  disabled=""
  type="submit"
>
  Upload
</button>
`;
